File: classes/parte-7-paradigmas/122-asincrono-async-await-y-promesas/implementaciones/java/Main.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.ExecutionException;

public class Main {
    public static void main(String[] args) throws IOException, InterruptedException, ExecutionException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine().trim());
        CompletableFuture<Integer> tarea = CompletableFuture.supplyAsync(() -> n * 2);
        System.out.println("resultado=" + tarea.get());
    }
}

File: classes/parte-7-paradigmas/122-asincrono-async-await-y-promesas/implementaciones/javascript/main.mjs
import { readFileSync } from "node:fs";

async function doble(x) {
  return x * 2;
}

const n = parseInt(readFileSync(0, "utf8").trim(), 10);
(async () => {
  const resultado = await doble(n);
  console.log(`resultado=${resultado}`);
})();
